src/Git/Repository.php

```diff
@@ -51,23 +51,31 @@ public function getCommitsLogPatch($file)
             . "<commiter_date>%ct</commiter_date>"
             . "<message><![CDATA[%s]]></message>"
             . "<body><![CDATA[%b]]></body>"
-            . "</item>\" -- \"$file\""
+            . "</item>\" --max-count=".($max_count+1)." $revisions -- \"$file\""
         );
 
         $patch_collection = array();
-        foreach ( preg_split('/('.$record_delimiter.'\<item\>)/', $file_patches,null, PREG_SPLIT_NO_EMPTY) as $patches) {
-            $patches = '<item>' . $patches;
-            $xmlEnd = strpos($patches, '</item>') + 7;
-            $commitInfo = substr($patches, 0, $xmlEnd);
-            $commitData = substr($patches, $xmlEnd);
+        $format = new PrettyFormat;
+        $commits = preg_split('/'.$record_delimiter.'(?:<item>)/', $file_patches, null, PREG_SPLIT_NO_EMPTY);
+        $file_patches = null;
+        $pagination = (count ($commits) == $max_count+1);
+        $count = 0;
+        foreach ($commits as $patch) {
+            $xmlEnd = strpos($patch, '</item>') + strlen ('</item>');
+            $commitInfo = "<item>".substr($patch, 0, $xmlEnd);
+            $commitData = substr($patch, $xmlEnd);
             $logs = explode("\n", $commitData);
 
             // Read commit metadata
-            $format = new PrettyFormat;
             $data = $format->parse($commitInfo);
             $commit = new Commit;
             $commit->importData($data[0]);
             $commit->setDiffs($this->readDiffLogs($logs));
+
+            // Limit number of commits due to memory constrains
+            $count++;
+            if ($pagination && ($count == $max_count+1)) { $commit->setShortHash(null); }
```

Comment:
PSR2 dictates this should be three independent lines. And `$max_count+1` should be `$max_count + 1`